Tasks/UseRubyVersion/userubyversion.ts

```diff
@@ -0,0 +1,101 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+
+import * as semver from 'semver';
+
+import * as task from 'vsts-task-lib/task';
+import * as tr from 'vsts-task-lib/toolrunner';
+import * as tool from 'vsts-task-tool-lib/tool';
+
+export enum Platform {
+    Windows,
+    MacOS,
+    Linux
+}
+
+export function getPlatform(): Platform {
+    switch (process.platform) {
+        case 'win32': return Platform.Windows;
+        case 'darwin': return Platform.MacOS;
+        case 'linux': return Platform.Linux;
+        default: throw Error(task.loc('PlatformNotRecognized'));
+    }
+}
+
+interface TaskParameters {
+    readonly versionSpec: string;
+    readonly outputVariable: string;
+    readonly addToPath: boolean;
+}
+
+function evaluateVersions(versions: semver.SemVer[], versionSpec: string): semver.SemVer | null {
```

Comment:
Same for this as well, should this be in the tool-lib or a common helper?

Reply:
This is exists in tool-lib.  Any reason not to use that implementation?

Reply:
No.  Unfortunately, I can't use the one in tool-lib.  I need it so that it accepts SemVer, because it's not a clean version that gets passed in.  It has to be `clean`ed first.  This was copied and adapted to work with the way the versions are pulled in from `ruby-switch --list`.

Reply:
Sorry, when I said `clean`ed, I actually meant `coerce`ed.  See line: 74.

Reply:
Sorry I don't follow.  I get that the output from `ruby-switch` might need some processing to turn them into full semantic versions (missing the 'patch' number perhaps?).  But you should be able to do that processing and then pass it to `evaluateVersions` as a list of strings -- e.g. `.map(x => x.toString()`?

Reply:
Since the keys of `versionMap` are the string representations of the coerced semver versions, it seems like `Object.keys(versionMap)` is exactly what we want here.